tests/main/snap-confine-privs/task.yaml

```diff
@@ -23,7 +23,9 @@ prepare: |
     echo "Compile and prepare the support program"
     # Because we use the snap data directory we don't need to clean it up
     # manually as all snaps and their data are reset after each test.
-    gcc -Wall -Wextra -Werror ./uids-and-gids.c -o "$P/uids-and-gids"
+    # Build the test binary statically, as it will be running inside a base with
+    # potentially older glibc.
+    gcc -Wall -Wextra -Werror ./uids-and-gids.c -o "$P/uids-and-gids" -static
```

Comment:
Minor consistency nitpick, maybe add -static at the same spot in both cases (last argument is probably better as it stands out more).